File: app/src/main/java/com/octosync/selfimprove/Habit.java
package com.octosync.selfimprove;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "habits")
public class Habit {
    public enum Status {
        PENDING, SUCCESS, FAILED
    }

    @PrimaryKey(autoGenerate = true)
    private int id;
    private String name;
    private Status status;

    public Habit(String name) {
        this.name = name;
        this.status = Status.PENDING;
    }

    // Room needs this
    public void setId(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Status getStatus() {
        return status;
    }

    public void setStatus(Status status) {
        this.status = status;
    }
}

File: app/src/main/java/com/octosync/selfimprove/Habit.kt
package com.octosync.selfimprove

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "habits")
data class Habit @JvmOverloads constructor(
    val name: String,
    var status: Status = Status.PENDING,
    val category: Category = Category.OTHER,
    val date: Long = System.currentTimeMillis() / (24 * 60 * 60 * 1000) * (24 * 60 * 60 * 1000), // Normalized to start of day
    val reminderTime: Long? = null, // Milliseconds from start of day
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0
) {
    enum class Status {
        PENDING, SUCCESS, FAILED
    }

    enum class Category {
        PHYSICAL, MENTAL, PRODUCTIVITY, OTHER
    }
}
